src/plugins/discover/public/embeddable/saved_search_embeddable.tsx

```diff
@@ -178,7 +182,7 @@ export class SavedSearchEmbeddable
         unwrapResult
       );
 
-      this.panelTitle = this.savedSearch.title ?? '';
+      this.panelTitle = this.getCurrentTitle();
```

Comment:
We were setting `panelTitle` incorrectly before which would cause the state comparison/reload logic to trigger when comparing to `output.title` if a custom title was used, even though nothing actually changed.

Reply:
Is line 151 still correct?

Reply:
I believe so -- `initializeOutput` calls `this.getCurrentTitle()` on line 208 and passes it to `this.updateOutput()` on line 225. Line 151 should sync `this.panelTitle` with the updated `output.title` if it detects a change. It's a bit convoluted tbh, but I think it's working as intended.